cmd/cogito/main.go

```diff
@@ -7,21 +7,38 @@ import (
 	"io"
 	"os"
 	"path"
+
+	"github.com/Pix4D/cogito/cogito"
+	"github.com/Pix4D/cogito/resource"
+	"github.com/hashicorp/go-hclog"
 )
 
 func main() {
 	// The "Concourse resource protocol" expects:
 	// - stdin, stdout and command-line arguments for the protocol itself
 	// - stderr for logging
 	// See: https://concourse-ci.org/implementing-resource-types.html
-	if err := run(os.Stdin, os.Stdout, os.Args); err != nil {
+	if err := run(os.Stdin, os.Stdout, os.Stderr, os.Args); err != nil {
 		fmt.Fprintln(os.Stderr, err)
 		os.Exit(1)
 	}
 }
 
-func run(in io.Reader, out io.Writer, args []string) error {
+func run(in io.Reader, out io.Writer, logOut io.Writer, args []string) error {
+	log := hclog.New(&hclog.LoggerOptions{
```

Comment:
This introduces https://github.com/hashicorp/go-hclog, which I used in other projects and consider good.